Projects/App/Sources/AppDelegate+Notifications.swift

```diff
@@ -42,6 +42,83 @@ extension AppDelegate {
 }
 
 extension AppDelegate: UNUserNotificationCenterDelegate {
+    fileprivate func performPostLoggedIn(work: @escaping () -> Void) {
+        bag += ApplicationContext.shared.$isLoggedIn.atOnce().filter { $0 }
+            .onFirstValue { _ in
+                work()
+            }
+    }
+
+    fileprivate func performPushAction(notificationType: String, userInfo: [AnyHashable: Any]) {
+        hAnalyticsEvent.notificationOpened(type: notificationType).send()
+
+        if notificationType == "NEW_MESSAGE" {
+            performPostLoggedIn {
+                let store: UgglanStore = globalPresentableStoreContainer.get()
+                store.send(.openChat)
```

Comment:
We need to add check if the chat screen is already opened, so we don't open it twice.
It will not open over the current chat, but will show again after we dismiss chat.

Reply:
Awesome! It works fine!